### PR TITLE
Fixed Part #4200 : Added dark mode support to AudioLanguageActivity, ReadingTextSizeActivity, AppLanguageActivity

### PR DESCRIPTION
**Pull request by MohitGupta121**
## Fixed One Part of #4200 : Added dark mode support to AudioLanguageActivity, ReadingTextSizeActivity, AppLanguageActivity

Dark Mode Mock:-

- ReadingTextSizeActivity:- https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/2eaf085a-c2a7-4b1d-b515-5b32f8642669/
- AppLanguageActivity:- https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/2b8387b0-5ec6-4e42-bffa-c039128b2f12
- AudioLanguageActivity:- https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/f8cf647c-85ea-4900-b98a-9449dd3a82a6/

|Default|Dark Mode|
|:----------------------------------------:|:-----------------------------------------:|
|  | Screenshot_2022-05-31-12-23-27-332_com android camera |

|Default|Dark Mode|
|:----------------------------------------:|:-----------------------------------------:|
|  |  |

|Default|Dark Mode|
|:----------------------------------------:|:-----------------------------------------:|
|  | Screenshot_2022-05-31-12-23-27-332_com android camera |

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Review by MohitGupta121 — COMMENTED** [review]
Done Dark Mode

File: app/src/main/res/values-land/dimens.xml [hunk 1]
```diff
@@ -539,4 +539,4 @@
 
   <!-- ProfileListFragment -->
   <dimen name="profile_list_fragment_recycler_view_padding_bottom">108dp</dimen>
-</resources>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 2]
```diff
@@ -503,4 +503,4 @@
   <dimen name="topic_fragment_tab_layout_margin_start">192dp</dimen>
   <dimen name="topic_fragment_tab_layout_margin_end">192dp</dimen>
   <dimen name="topic_fragment_tab_layout_tab_indicator_height">4dp</dimen>
-</resources>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 3]
```diff
@@ -502,4 +502,4 @@
   <dimen name="topic_fragment_tab_layout_margin_start">64dp</dimen>
   <dimen name="topic_fragment_tab_layout_margin_end">64dp</dimen>
   <dimen name="topic_fragment_tab_layout_tab_indicator_height">4dp</dimen>
-</resources>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/values-xxhdpi/dimens.xml [hunk 4]
```diff
@@ -5,4 +5,4 @@
   <dimen name="profile_chooser_fragment_profile_recycler_view_margin_top">36dp</dimen>
   <dimen name="profile_chooser_profile_view_view_margin_top">60dp</dimen>
   <dimen name="profile_chooser_add_view_margin_top_profile_not_added">60dp</dimen>
-</resources>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values-xxhdpi/dimens.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/values/component_colors.xml [hunk 5]
```diff
@@ -160,4 +160,11 @@
   <color name="component_color_math_expression_parser_activity_radio_buttons_color">@color/color_palette_dark_text_color</color>
   <color name="component_color_math_expression_parser_activity_edit_text_color">@color/color_palette_dark_text_color</color>
   <color name="component_color_math_expression_parser_activity_hint_text_color">@color/color_palette_hint_text_color</color>
+  <!-- Audio Language Activity -->
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml:208]: Please add similar comments for reading text size and app language related actvities.

File: app/src/main/res/values/dimens.xml [hunk 6]
```diff
@@ -748,4 +748,4 @@
 
   <!-- ProfileListFragment -->
   <dimen name="profile_list_fragment_recycler_view_padding_bottom">40dp</dimen>
-</resources>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: EOF is missing, add one blank line at the end.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@MohitGupta121 PTAL

File: app/src/main/res/values/component_colors.xml (hunk 5 above)
Comment (MohitGupta121) [reply app/src/main/res/values/component_colors.xml:163]: Actually I added that at line 168.

**Review by MohitGupta121 — COMMENTED** [review]
Done with requested changes

**Comment by MohitGupta121** [comment]
@rt4914 can you please review this PR

**Comment by rt4914** [comment]
@MohitGupta121 Please don't resolve comments on your own. The person who has started the comment thread will resolve them. Instead you should reply to those comments so that the owner can device whether to close the conversation or not.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@MohitGupta121 The colors in general do not match with the mocks.

1. The highlighted selection color is different it should be `#00A89F`
2. The text color is different it should be `#FFFFFF`


Also it looks like even the background color, the divider color is different.


Please have a look at each and every color and use it correctly.

**Comment by MohitGupta121** [comment]
> 

Okay I'll change that.

**Comment by MohitGupta121** [comment]
> @MohitGupta121 Please don't resolve comments on your own. The person who has started the comment thread will resolve them. Instead you should reply to those comments so that the owner can device whether to close the conversation or not.

Oo okay sorry I'll take care next time

File: app/src/main/res/values/component_colors.xml [hunk 7]
```diff
@@ -186,4 +186,13 @@
   <color name="component_color_math_expression_parser_activity_radio_buttons_color">@color/color_palette_dark_text_color</color>
   <color name="component_color_math_expression_parser_activity_edit_text_color">@color/color_palette_dark_text_color</color>
   <color name="component_color_math_expression_parser_activity_hint_text_color">@color/color_palette_hint_text_color</color>
+    <!-- Audio Language Activity -->
+  <color name="component_color_audio_language_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_language_items_background_color">@color/color_palette_container_background_color</color>
+  <color name="component_color_language_items_heading_text_color">@color/color_palette_dark_text_color</color>
+  <!-- Reading Text Size Activity -->
+  <color name="component_color_reading_text_size_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_reading_text_size_items_heading_text_color">@color/color_palette_dark_text_color</color>
```
Comment (MohitGupta121) [inline app/src/main/res/values/component_colors.xml:214]: @rt4914 actually the text color is already White `#FFFFFF` , due to my Phone brightness is low that's why it's looking fade in the screenshot.

For the selected color, `#00A89F` I'm changing that one.

Comment (rt4914) [reply app/src/main/res/values/component_colors.xml:214]: I don't think that brightness affects the screenshot. How are you taking screenshots?

Comment (MohitGupta121) [reply app/src/main/res/values/component_colors.xml:214]: Simply as by default screenshot option in Android.

Comment (MohitGupta121) [reply app/src/main/res/values/component_colors.xml:214]: > 

As it's showing white to me when I'm testing, also in code the color is white and light green for selection.

https://user-images.githubusercontent.com/76530270/180618911-1fd4171a-999a-4c3f-8625-72b9179bc4b0.jpg

Comment (MohitGupta121) [reply app/src/main/res/values/component_colors.xml:214]: > 

@rt4914 can you please check this one? Is it okay?

https://user-images.githubusercontent.com/76530270/181081446-24ae39b1-e752-45dc-a2c6-5470881d48c0.jpg

Comment (rt4914) [reply app/src/main/res/values/component_colors.xml:214]: Actually it does not look correct still. I will try to run this locally and see why the screenshots are coming in different colors.

**Comment by rt4914** [comment]
@MohitGupta121 Please keep in mind that you assign your PRs to me if you need me to check something otherwise I won't be able to see your PRs.

**Comment by MohitGupta121** [comment]
> @MohitGupta121 Please keep in mind that you assign your PRs to me if you need me to check something otherwise I won't be able to see your PRs.

Okay, I'll take care next time.

**Comment by BenHenning** [comment]
As discussed @MohitGupta121 can you upload new screenshots that use a dark mode emulator? It may be something odd with the physical device or specific Android OEM version (i.e. maybe it's performing some sort of extra shading).

**Comment by MohitGupta121** [comment]
> 

Okay, I'll update it in a few hours.

**Comment by MohitGupta121** [comment]
@BenHenning @rt4914 I updated the screenshots through emulator dark mode. Can you please review it now. Is it okay?

**Comment by BenHenning** [comment]
Thanks @MohitGupta121. To confirm: do the individual rows appear as dark on the emulator as they do in the screenshots you added?

**Comment by MohitGupta121** [comment]
> Thanks @MohitGupta121. To confirm: do the individual rows appear as dark on the emulator as they do in the screenshots you added?

Yes they all are the same in the emulator. It's look same as in dark mode mockups.

File: app/src/main/res/values/component_colors.xml (hunk 7 above)
Comment (rt4914) [reply app/src/main/res/values/component_colors.xml:214]: Now it looks correct.

File: app/src/main/res/values/component_colors.xml [hunk 8]
```diff
@@ -205,4 +205,13 @@
...
+  <!-- Audio Language Activity -->
+  <color name="component_color_audio_language_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_language_items_background_color">@color/color_palette_container_background_color</color>
+  <color name="component_color_language_items_heading_text_color">@color/color_palette_dark_text_color</color>
+  <!-- Reading Text Size Activity -->
+  <color name="component_color_reading_text_size_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_reading_text_size_items_heading_text_color">@color/color_palette_dark_text_color</color>
+  <!-- App Language Activity -->
+  <color name="component_color_app_language_activity_background_color">@color/color_palette_dark_background_color</color>
+</resources>
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: EOF is missing, add one blank line at the end.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Nice work @MohitGupta121 Just one suggestion.

File: app/src/main/res/values/component_colors.xml (hunk 8 above)
Comment (MohitGupta121) [reply app/src/main/res/values/component_colors.xml]: okay sure, thanks you.

Comment (MohitGupta121) [reply app/src/main/res/values/component_colors.xml]: @rt4914 can you please check it now I added EOF.

**Comment by MohitGupta121** [comment]
Suddenly 2 test case are failing. Unable to find exact reason. I only added EOF. It's due to this https://github.com/oppia/oppia-android/runs/8001170895?check_suite_focus=true#step:7:1764

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Comment by rt4914** [comment]
@MohitGupta121 Please don't `Resolve Conversations`. The conversation should be resolved by the person who started that thread.

**Comment by MohitGupta121** [comment]
> @MohitGupta121 Please don't `Resolve Conversations`. The conversation should be resolved by the person who started that thread.

Oh okay 😅 sorry once again take care next time.

**Pull request merged**